### ISSUE TITLE
Non-text in text fields of miniSEED 2 blockettes causes record parsing failure 

### ISSUE THREAD

**Issue by chad-iris** [iris-edu/libmseed#59]
If a MS2 blockette's text field is mapped to an extra header and contains non-UTF-8 characters, parsing will fail with messages like:

```
ERROR: mseh_add_event_detection(): Cannot set: Detector
ERROR: msr3_unpack_mseed2(XFDSN:GT_BDFB__B_H_Z): Problem mapping Blockette 201 to extra headers
ERROR: Cannot read TEST-B201-bad-detector-name.mseed: No miniSEED data detected
```

Test data is attached:
[TEST-B201-bad-detector-name.mseed.zip](https://github.com/iris-edu/libmseed/files/3468571/TEST-B201-bad-detector-name.mseed.zip)

**Comment by chad-iris** [comment]
Note that non-UTF-8 characters are not allowed in JSON and generally not printable so there is no need to support any such byte strings in text fields.

**Comment by chad-iris** [comment]
Fixed in 1cfa1ea8f065952c78d25b60d99bb99d0b310ff4 and 669a0faed8457a53a3c71fdbd4ed732a9a448c4d.

Now only UTF-8 characters are copied by the libraries string copy routines.  This avoids copying unprintable garbage during parsing.
